src/main/features/claim/routes/confirmation.ts

```diff
@@ -3,12 +3,14 @@ import { Paths } from 'claim/paths'
 
 import ClaimStoreClient from 'claims/claimStoreClient'
 import Claim from 'claims/models/claim'
+import OwnershipChecks from 'app/auth/ownershipChecks'
 
 export default express.Router()
   .get(Paths.confirmationPage.uri, async (req: express.Request, res: express.Response, next: express.NextFunction) => {
     const { externalId } = req.params
     try {
       const claim: Claim = await ClaimStoreClient.retrieveByExternalId(externalId)
+      OwnershipChecks.checkClaimOwner(res.locals.user, claim)
```

Comment:
This is not the right approach.
`ClaimStoreClient.retrieveByExternalId` should have the check and there need's to be a forbidden page explaining what happened

Reply:
I would just slightly change the approach to not add the ownership check logic to `ClaimStoreClient.retrieveByExternalId` and creating `ClaimStoreService` instead which delegates data retrieval to generic client and throws an error when ownership check failed.

That would keep HTTP client simple with one responsibility only and in sync with the API which does not have any ownership logic.